src/Polly.Core/Hedging/OnHedgingArguments.cs

```diff
@@ -14,36 +14,31 @@ namespace Polly.Hedging;
     /// <summary>
     /// Initializes a new instance of the <see cref="OnHedgingArguments{TResult}"/> struct.
     /// </summary>
-    /// <param name="outcome">The context in which the resilience operation or event occurred.</param>
-    /// <param name="context">The outcome of the resilience operation or event.</param>
+    /// <param name="primaryContext">The outcome of the resilience operation or event.</param>
+    /// <param name="actionContext">The action context.</param>
     /// <param name="attemptNumber">The zero-based hedging attempt number.</param>
-    /// <param name="duration">The execution duration of hedging attempt or the hedging delay in case the attempt was not finished in time.</param>
-    public OnHedgingArguments(ResilienceContext context, Outcome<TResult>? outcome, int attemptNumber, TimeSpan duration)
+    public OnHedgingArguments(ResilienceContext primaryContext, ResilienceContext actionContext, int attemptNumber)
     {
-        Context = context;
-        Outcome = outcome;
+        PrimaryContext = primaryContext;
+        ActionContext = actionContext;
         AttemptNumber = attemptNumber;
-        Duration = duration;
     }
 
     /// <summary>
-    /// Gets the outcome that needs to be hedged, if any.
+    /// Gets the primary resilience context.
     /// </summary>
-    /// <remarks>If this property is <see langword="null"/>, it's an indication that user-callback or hedged operation did not complete within the hedging delay.</remarks>
-    public Outcome<TResult>? Outcome { get; }
+    public ResilienceContext PrimaryContext { get; }
 
     /// <summary>
-    /// Gets the context of this event.
+    /// Gets the action context that will be used for the hedged action.
     /// </summary>
-    public ResilienceContext Context { get; }
+    /// <remarks>
+    /// This context is cloned from <see cref="PrimaryContext"/>.
+    /// </remarks>
+    public ResilienceContext ActionContext { get; }
 
     /// <summary>
     /// Gets the zero-based hedging attempt number.
```

Comment:
Is this the correct explanation? Looking at this fresh, if this is zero-based shouldn't we call it an index rather than a number?

Reply:
IF you don't mind i'll leave it as it is because we agreed on this in the API review and to me it sounds nicer that `AttemptIndex`.

Reply:
Ok, but I would expect the first value to be one if it's a number. I remember discussing calling it `Number` but I thought that was in the context of retries and it wasn't zero-based.

Reply:
For retries it's also called `AttemptNumber` and it's also zero-based. It's just the name we agreed on. Docs make it clear it starts from zero.